python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_defs.py

```diff
@@ -88,3 +92,208 @@ def _assets(context):
+
+        return tables
+
+
+class AirbyteSource(NamedTuple("_AirbyteSource", [("name", str)])):
+    """
+    Contains information about an Airbyte source.
+
+    Attributes:
+        name (str): The name of the source.
+    """
+
+    @classmethod
+    def from_config(cls, contents: Mapping[str, Any]) -> "AirbyteSource":
+        return cls(name=contents["resource_name"])
+
+
+@experimental
+def load_assets_from_airbyte_project(
+    project_dir: str,
+    key_prefix: Optional[CoercibleToAssetKeyPrefix] = None,
+    source_key_prefix: Optional[CoercibleToAssetKeyPrefix] = None,
+    create_assets_for_normalization_tables: bool = True,
+    connection_to_group_fn: Optional[Callable[[str], Optional[str]]] = _clean_name,
+) -> List[Union[AssetsDefinition]]:
+    """
+    Loads an Airbyte project into a set of Dagster assets.
+
+    Point to the root folder of an Airbyte project synced using the Octavia CLI. For
+    more information, see https://github.com/airbytehq/airbyte/tree/master/octavia-cli#octavia-import-all.
+
+    Args:
+        project_dir (str): The path to the root of your Airbyte project, containing sources, destinations,
+            and connections folders.
+        key_prefix (Optional[CoercibleToAssetKeyPrefix]): A prefix for the asset keys created.
+        source_key_prefix (Optional[CoercibleToAssetKeyPrefix]): A prefix for the source asset keys produced.
+        create_assets_for_normalization_tables (bool): If True, assets will be created for tables
+            created by Airbyte's normalization feature. If False, only the destination tables
+            will be created.
+        connection_to_group_fn (Optional[Callable[[str], Optional[str]]]): Function which returns an asset
+            group name for a given Airbyte connection name. If None, no groups will be created. Defaults
+            to a basic sanitization function.
+    """
+
+    if isinstance(source_key_prefix, str):
+        source_key_prefix = [source_key_prefix]
+    source_key_prefix = check.list_param(source_key_prefix or [], "source_key_prefix", of_type=str)
+
+    if isinstance(key_prefix, str):
+        key_prefix = [key_prefix]
+    key_prefix = check.list_param(key_prefix or [], "key_prefix", of_type=str)
+
+    assets: List[AssetsDefinition] = []
+
+    connections_dir = os.path.join(project_dir, "connections")
+    for connection_name in os.listdir(connections_dir):
+        connection_dir = os.path.join(connections_dir, connection_name)
+        with open(os.path.join(connection_dir, "configuration.yaml"), encoding="utf-8") as f:
+            connection = AirbyteConnection.from_config(yaml.safe_load(f.read()))
+
+        with open(os.path.join(project_dir, connection.source_config_path), encoding="utf-8") as f:
+            source = AirbyteSource.from_config(yaml.safe_load(f.read()))
+            source_asset_key = AssetKey(source_key_prefix + [_clean_name(source.name)])
+
+        state_file = next(
```

Comment:
will there ever be multiple state files in the directory under normal circumstances? or does the cli always delete previous versions when writing a new one?

Reply:
I think there could be multiple if the user has multiple workspaces - I'll add an optional workspace ID argument which will disambiguate & error if a user doesn't provide it & there's more than one state file.

Reply:
did the above